tests/test_process.py

```diff
@@ -796,7 +797,69 @@ async def test():
 
 
 class Test_UV_Process(_TestProcess, tb.UVTestCase):
-    pass
+    def test_process_preexec_fn_double_close(self):
+        script = textwrap.dedent("""
+            import os
+            import sys
+            import threading
+            import queue
+            import concurrent.futures
+
+            pid = os.getpid()
+            q = queue.Queue()
+            evt = threading.Event()
+            r, w = os.pipe()
+            pipe = os.pipe
+            close = os.close
+
+
+            def mock_pipe():
+                rv = pipe()
+                q.put(rv[1])
+                return rv
+
+
+            def mock_close(fd):
+                close(fd)
+                if os.getpid() == pid:
+                    q.put(fd)
+                    evt.wait()
+
+
+            os.pipe = mock_pipe
+            os.close = mock_close
+
+            import asyncio
+            import uvloop
+
+            uvloop.install()
```

Comment:
When the 3.11 test suite lands it might be worth conditionally using the modern `asyncio.Runner(loop_factory=uvloop.new_event_loop)`